Src/java/cql-to-elm/src/test/java/org/cqframework/cql/cql2elm/SemanticTests.java

```diff
@@ -652,6 +653,62 @@ public void testNonExistentFileName() {
         assertThrows(IOException.class, () -> TestUtils.runSemanticTest("ThisFileDoesNotExist.cql", 0));
     }
 
+    @Test
+    public void testCaseConditionalReturnTypes() throws IOException {
+        CqlTranslator translator = runSemanticTest("Issue648.cql", 0);
+        org.hl7.elm.r1.Library library = translator.toELM();
+        Map<String, ExpressionDef> defs = new HashMap<>();
+
+        if (library.getStatements() != null) {
+            for (ExpressionDef def : library.getStatements().getDef()) {
+                defs.put(def.getName(), def);
+            }
+        }
+
+        ExpressionDef caseDef = defs.get("Cases");
+    
+        assertThat(caseDef.getResultType(), instanceOf(ChoiceType.class));
+
+        ChoiceType choiceType = (ChoiceType)caseDef.getResultType();
+
+        List<String> expectedChoiceTypes = new ArrayList<>();
+        expectedChoiceTypes.add("System.String");
+        expectedChoiceTypes.add("System.Boolean");
+        expectedChoiceTypes.add("System.Integer");
+
+        for (DataType dt : choiceType.getTypes()) {
+                assertTrue("Expected types are String, Boolean, and Integer: ", expectedChoiceTypes.contains(((NamedType)dt).getName()));
+        }
+        
+    }
+
+    @Test
+    public void testIfConditionalReturnTypes() throws IOException {
+        CqlTranslator translator = runSemanticTest("Issue648.cql", 0);
+        org.hl7.elm.r1.Library library = translator.toELM();
+        Map<String, ExpressionDef> defs = new HashMap<>();
+
+        if (library.getStatements() != null) {
+            for (ExpressionDef def : library.getStatements().getDef()) {
+                defs.put(def.getName(), def);
+            }
+        }
+
+        ExpressionDef ifDef = defs.get("If");
+        assertThat(ifDef.getResultType(), instanceOf(ChoiceType.class));
+
+        ChoiceType choiceType = (ChoiceType)ifDef.getResultType();
+
+        List<String> expectedChoiceTypes = new ArrayList<>();
+        expectedChoiceTypes.add("System.String");
+        expectedChoiceTypes.add("System.Boolean");
+
+        for (DataType dt : choiceType.getTypes()) {
+                assertTrue("Expected return types are String and Boolean: ", expectedChoiceTypes.contains(((NamedType)dt).getName()));
```

Comment:
Suggestion: Same as above here. The set of expected types should be equivalent to the set of actual types

Reply:
The requested change have been made.